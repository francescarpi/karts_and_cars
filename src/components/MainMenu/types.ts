export interface IMenuItem {
  name: string;
  type: string;
  href: string;
}
