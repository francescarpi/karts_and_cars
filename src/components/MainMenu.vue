<script setup lang="ts"></script>

<template>
  <div class="bg">
    <div class="bg-black text-white p-4 opacity-80">Menu</div>
  </div>
</template>

<style scoped lang="scss">
.bg {
  background-image: url("@/assets/imgs/flag.png");
}
</style>
