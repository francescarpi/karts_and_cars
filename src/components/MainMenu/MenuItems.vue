<script setup lang="ts">
defineProps({});
</script>

<template>Items</template>
