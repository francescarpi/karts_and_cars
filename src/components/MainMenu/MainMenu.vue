<script setup lang="ts">
import MenuItems from "./MenuItems.vue";
import { typeHash } from "@/constants";
import { Bars3Icon } from "@heroicons/vue/24/outline/";
import type { IMenuItem } from "./types";

const options: IMenuItem[] = [
  {
    name: "Identidad",
    type: typeHash,
    href: "identity",
  },
  {
    name: "Galer√≠a",
    type: typeHash,
    href: "gallery",
  },
  {
    name: "Servicios",
    type: typeHash,
    href: "services",
  },
  {
    name: "Eventos",
    type: typeHash,
    href: "events",
  },
  {
    name: "Contacto",
    type: typeHash,
    href: "contact",
  },
];
</script>

<template>
  <div class="bg">
    <div class="bg-black text-white p-4 opacity-80">
      <div class="hidden md:max-w-6xl md:mx-auto md:block">
        <MenuItems :options="options" />
      </div>
      <Bars3Icon class="block md:hidden w-6" />
    </div>
  </div>
</template>

<style scoped lang="scss">
.bg {
  background-image: url("@/assets/imgs/flag.png");
}
</style>
