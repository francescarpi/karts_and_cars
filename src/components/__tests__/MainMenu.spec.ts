import MainMenu from "../MainMenu/MainMenu.vue";
import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";

describe("MainMenu", () => {
  it("renders properly", () => {
    const wrapper = mount(MainMenu);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
