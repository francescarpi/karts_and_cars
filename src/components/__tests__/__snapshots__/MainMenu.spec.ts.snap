// Vitest Snapshot v1

exports[`MainMenu > renders properly 1`] = `
"<div class=\\"bg\\" data-v-43bf210c=\\"\\">
  <div class=\\"bg-black text-white p-4 opacity-80\\" data-v-43bf210c=\\"\\">Menu</div>
</div>"
`;
